gluten-core/src/main/scala/io/glutenproject/execution/TakeOrderedAndProjectExecTransformer.scala

```diff
@@ -82,19 +83,16 @@ case class TakeOrderedAndProjectExecTransformer(limit: Int,
       }
       val transformStageCounter: AtomicInteger =
         ColumnarCollapseTransformStages.transformStageCounter
-      val finalLimitPlan = if (childRDDPartsNum == 1) {
-          limitExecPlan
```

Comment:
This is interesting, it seems ok to remove a shuffle if the input rdd only has one partition. Why does it not work ?

Reply:
Lack of knowledge, I don't know why.